src/oop/static_modifier/medium/q5/MainTest.java

```diff
@@ -0,0 +1,113 @@
+        String classString = "oop.static_modifier.medium.q5." + className;
+        testClass = new ObjectTest(classString);
+        String outputClassString = "oop.static_modifier.medium.q5." + testClassName;
+        outputClass = new ObjectTest(outputClassString);
+        String modificationErrorMessage = "You have modified the provided portions of " + className + " class for %s attribute. Please revert them to the original state.";
+        assertTrue(testClass.hasField(attributeName, String.class),
+                String.format(modificationErrorMessage, attributeName));
+        assertTrue(testClass.hasModifier(attributeName, "private"),
+                String.format(modificationErrorMessage, attributeName));
+        String[] test = new String[]{"Black", "Blue", "Red", "Brown"};
+        for (int i = 0; i < test.length; i++) {
+            Object[][] arguments = {
+                    {test[i], String.class}
+            };
+            Object classInstance = testClass.createInstance(arguments);
+            _assertEquals(test[i], testClass.getFieldValue(classInstance, attributeName),
+                    String.format(modificationErrorMessage, attributeName));
+        }
+    }
+
+    @Test
+    @Tag("dependency1")
+    public void correctTestClass() throws Throwable {
+        Object classInstance = outputClass.createInstance();
+        String[] methodModifiers = {"public"};
+        Object[][] arguments = {
+                {new String[0], String[].class}
+        };
+        outputClass.callMethod("main", arguments, methodModifiers, classInstance, new Clause[]{
+                new StringLiteral("This is a " + initialName + " that is " + initialColour + " in colour"),
+                new Optional(new StringLiteral(" ")),
+                new Optional(new NewLine())
+        });
+    }
+
+    @Test
+    public void hatClassHasCorrectAttributes() {
+        String incorrectNameError = "Your %s class does not include the %s attribute.";
+        String incorrectVisibilityError = "Your %s attribute does not have the correct visibility modifier.";
+        String incorrectStaticError = "Your %s attribute does not have the static modifier.";
+        assertTrue(testClass.hasField(staticAttributeName, String.class),
+                String.format(incorrectNameError, className, staticAttributeName));
+        assertTrue(testClass.hasModifier(staticAttributeName, "private"),
+                String.format(incorrectVisibilityError, staticAttributeName));
+        assertTrue(testClass.hasModifier(staticAttributeName, "static"),
+                String.format(incorrectStaticError, staticAttributeName));
+    }
+
+    private static Stream<String> inputProvider() {
+        return Stream.of(
+                "Black", "Blue", "Red", "Brown"
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    @Tag("dependent1")
+    public void correctToStringMethod(String colour) throws Throwable {
```

Comment:
You should test this with different company names.

Reply:
To do this you will also probably have to order the tests. See #694 for an example if you need it.